pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -640,4 +640,11 @@ def determine_root_gpu_device(gpus):
     # set root gpu
     root_gpu = gpus[0]
 
+    # set cuda device to root gpu
+    # related to https://github.com/PyTorchLightning/pytorch-lightning/issues/958
+    # Refer solution: https://github.com/pytorch/pytorch/issues/9871#issuecomment-408304190
+    # root_device = torch.device("cuda", root_gpu)
+    root_device = (torch.device("cuda", root_gpu) if root_gpu >= 0 else torch.device("cpu"))
```

Comment:
What would root_device be set if the user wants CPU? None? -1? Maybe we should check for that explicitly

Reply:
If the user wants CPU, the function `determine_root_gpu_device` should not be called (?)

Reply:
in this case it is getting called with gpus=None, and returns None (see first lines of determine_root_gpu_device). So your `else torch.device("cpu"))` is never relevant.
I don't see anything wrong with just 
`root_device = torch.device("cuda", root_gpu)`

Reply:
but I think the device should be set outside this function anyway

Reply:
@awaelchli where do you suggest?

Reply:
I would search the code base for occurrences of `self.root_gpu` and check whether it is needed to set the device in each case. 
Maybe consider setting directly after [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/a707d4bea1a78a98265fd1ea5b7a7a6cadc37fb9/pytorch_lightning/trainer/trainer.py#L385) but I am not sure if that's the best place.

Reply:
better to ask the core team on this :)